### PR TITLE
CORE-105 Sprint Submission & Review forms

### PR DESCRIPTION
**Pull request by mellogarrett**
## Jira Issue

[CORE-105](https://lambdaschool.atlassian.net/browse/CORE-339?atlOrigin=eyJpIjoiNDc0NjJmYzQyMjE5NGQ0NGIyNjJlYjFhMmMwOWI2NGQiLCJwIjoiaiJ9)

## Description

- Add new Sprint Submission & Review forms
- Sprint Submission form is being server-rendered and is working end-to-end 🎉 

Sprint Submission Review form will be adjusted once the backend work for it has been completed.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: forms/sprint-submission-review.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    {
+      title: 'Anything else Lambda should know about this student?',
+      description: 'This feedback is internal only and will not be shared directly with the student.',
+      type: QuestionType.LongText,
+      isRequired: false,
+      externalKey: 'otherNotesInternal',
+    },
+    {
+      title: 'Please tag any specifics you observed during this sprint.',
+      type: QuestionType.Dropdown, // TODO: multi-select
```
Comment (MarkDKoslow) [inline forms/sprint-submission-review.ts]: Just a reminder to update this to multi-select

File: forms/sprint-submission-review.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        { key: 'after-hours', value: 'Attended After Hours', checked: false },
+        { key: 'job-search', value: 'Job Search', checked: false },
+        { key: 'warmup', value: 'Completed Warmup', checked: false },
+      ],
+      isRequired: true,
+      externalKey: 'tags',
+    },
+    {
+      title: 'What intervention or support, if any, did you recommend for this student?',
+      type: QuestionType.Dropdown, // TODO: multi-select
```
Comment (MarkDKoslow) [inline forms/sprint-submission-review.ts]: Same here

File: forms/sprint-submission-review.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      type: QuestionType.LongText,
+      isRequired: false,
+      externalKey: 'otherNotes',
+    },
+    {
+      title: 'Anything else Lambda should know about this student?',
+      description: 'This feedback is internal only and will not be shared directly with the student.',
+      type: QuestionType.LongText,
+      isRequired: false,
+      externalKey: 'otherNotesInternal',
```
Comment (MarkDKoslow) [inline forms/sprint-submission-review.ts]: Looks like this is just 'otherInternal' in the db (though i like otherNotesInternal more)

File: forms/sprint-submission-review.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      type: QuestionType.LongText,
+      isRequired: false,
+      externalKey: 'whatWentPoorly',
+    },
+    {
+      title: 'Describe this student in three words',
+      description: 'Imagine you are being asked to anonymously describe this student to a potential employer.',
+      type: QuestionType.LongText,
+      isRequired: false,
+      externalKey: 'studentInThreeWords',
```
Comment (MarkDKoslow) [inline forms/sprint-submission-review.ts]: Looks like this is `threeWordSummary` in the db

File: forms/sprint-submission-review.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  title: FormTitle.SprintSubmissionReview,
+  backends: [],
+  questions: [
+    {
+      title: 'Your Name',
+      type: QuestionType.Dropdown,
+      isRequired: true,
+      prepopulateChoices: true,
+      choices: [],
+      externalKey: 'studentId',
```
Comment (MarkDKoslow) [inline forms/sprint-submission-review.ts]: Looks like this is `projectManagerId`

File: forms/sprint-submission.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
+import { Form, Backend, QuestionType } from '../types';
+import { FormTitle } from '../constants/form-title';
+import { Validator } from '../constants/validator';
+
+export const SprintSubmissionForm: Form = {
+  title: FormTitle.SprintSubmission,
+  backends: [Backend.Zapier, Backend.FormsApi],
```
Comment (MarkDKoslow) [inline forms/sprint-submission.ts]: Can remove Zapier here

File: http/merge-question-options/atlas/index.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ export const mergeQuestionOptionsFromAtlas = async (form: Form): Promise<Form> =
   try {
     prepopulateOptions = (await axiosClient.get(getPrepopulateUrl(title))).data;
   } catch (err) {
-    isomorphicLogError(new LError('Error getting pre-populate options from Atlas', err));
+    isomorphicLogError(new LError('Error getting pre-populate options', err));
```
Comment (MarkDKoslow) [inline http/merge-question-options/atlas/index.ts]: good call 👍

File: forms/sprint-submission-review.ts (hunk 5 above)
Comment (MarkDKoslow) [reply forms/sprint-submission-review.ts]: (Can also merge these externalKey updates when we add the sprint sub review urls)

File: forms/sprint-submission-review.ts (hunk 1 above)
Comment (mellogarrett) [reply forms/sprint-submission-review.ts]: Oh duh, this can be a checkbox component... I was thinking a multi-select dropdown. Whoops!

**Comment by mellogarrett** [comment]
Thanks @MarkDKoslow, I think I've resolved all of your points.

**Pull request merged**